File: backend/src/main/scala/microservices/blog/objects/UpdateBlogPostPlannerRequest.scala
// UpdateBlogPostPlannerRequest：博客域更新博客文章请求对象。

package com.typesafe.travel.blog.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class UpdateBlogPostPlannerRequest(
    postId: String,
    userId: String,
    title: String,
    summary: String,
    content: String,
    images: List[BlogImagePlannerResponse],
    tags: Option[List[BlogTagPlannerResponse]],
    coverText: Option[String],
    travelCity: Option[String],
    travelCities: Option[List[String]]
)
object UpdateBlogPostPlannerRequest:
  given sourceEncoder: Encoder[UpdateBlogPostPlannerRequest] = deriveEncoder
  given sourceDecoder: Decoder[UpdateBlogPostPlannerRequest] = deriveDecoder

File: frontend/src/microservices/blog/objects/UpdateBlogPostPlannerRequest.ts
// UpdateBlogPostPlannerRequest：博客域更新博客文章请求对象。

import type { ContentImagePlannerResponse } from '@/microservices/content/objects/ContentImagePlannerResponse'
import type { BlogTagResponse } from './BlogTagResponse'

export type UpdateBlogPostPlannerRequest = {
  postId: string
  userId: string
  title: string
  summary: string
  content: string
  images: ContentImagePlannerResponse[]
  tags?: BlogTagResponse[]
  coverText?: string
  travelCity?: string | null
  travelCities?: string[] | null
}

export const updateBlogPostPlannerRequestFromJson = (json: string): UpdateBlogPostPlannerRequest =>
  JSON.parse(json) as UpdateBlogPostPlannerRequest

export const updateBlogPostPlannerRequestToJson = (value: UpdateBlogPostPlannerRequest): string =>
  JSON.stringify(value)